Source/WebCore/style/StyleBuilderConverter.h

```diff
@@ -1281,6 +1282,53 @@ inline GridAutoFlow BuilderConverter::convertGridAutoFlow(BuilderState&, const C
     return autoFlow;
 }
 
+inline OptionSet<MasonryAutoFlow> BuilderConverter::convertMasonryAutoFlow(BuilderState&, const CSSValue& value)
+{
+    auto& valueList = downcast<CSSValueList>(value);
+    if (!valueList.size())
+        return RenderStyle::initialMasonryAutoFlow();
+
+    auto* firstValue = downcast<CSSPrimitiveValue>(valueList.item(0));
+    auto* secondValue = downcast<CSSPrimitiveValue>(valueList.length() == 2 ? valueList.item(1) : nullptr);
+
+    OptionSet<MasonryAutoFlow> masonryAutoFlow;
+    switch (firstValue->valueID()) {
+    case CSSValuePack:
+        masonryAutoFlow.add(MasonryAutoFlow::Pack);
+        if (secondValue && secondValue->valueID() == CSSValueOrdered)
+            masonryAutoFlow.add(MasonryAutoFlow::Ordered);
+        else
+            masonryAutoFlow.add(MasonryAutoFlow::DefiniteFirst);    
+        break;
+    case CSSValueNext:
+        masonryAutoFlow.add(MasonryAutoFlow::Next);
+        if (secondValue && secondValue->valueID() == CSSValueOrdered)
+            masonryAutoFlow.add(MasonryAutoFlow::Ordered);
+        else
+            masonryAutoFlow.add(MasonryAutoFlow::DefiniteFirst);
+        break;
+    case CSSValueOrdered:
+        masonryAutoFlow.add(MasonryAutoFlow::Ordered);
+        if (secondValue && secondValue->valueID() == CSSValueNext)
+            masonryAutoFlow.add(MasonryAutoFlow::Next);
+        else
+            masonryAutoFlow.add(MasonryAutoFlow::Pack);
+        break;
+    case CSSValueDefiniteFirst:
+        masonryAutoFlow.add(MasonryAutoFlow::DefiniteFirst);
+        if (secondValue && secondValue->valueID() == CSSValueNext)
```

Comment:
This condition isn't possible is it? The only time the parser should have stored an ordered/definite-first value first is if we omitted the implicit 'pack' (which only happens if it was 'Ordered').

Could this be something like:

```
ASSERT(valueList.length() == 1 || 2)
if (valueList.length() == 2) {
    ASSERT(valueList[0].valueID() == Pack || Next);
    ASSERT(valueList[1].validID() == Ordered);
    return { valueList[0].valueID(), Ordered };
} else if (valueList[0].valueID() == CSSValuePack ||  valueList[0].valueID() == CSSValueNext) {
    return { valueList[0].valueID(), DefiniteFirst };
} else {
    ASSERT(valueList[0].validID() == Ordered);
    return { Pack, Ordered };
}
```

I find a lot of the CSS property code complicated since we have a value object that could be an arbitrarily complicated tree of other values, and there's no compile time verification of what 'valid' values are. Trying to structure code that makes it clear what the set of possible values are, with asserts, makes this easier to follow in my opinion :)

Reply:
Agreed I don't think it is possible either. Removed and tried to clean up the logic to make it a little more readable